src/core/ext/transport/chttp2/client/chttp2_connector.c

```diff
@@ -136,6 +136,8 @@ static void start_handshake_locked(grpc_exec_ctx *exec_ctx,
   c->handshake_mgr = grpc_handshake_manager_create();
   grpc_handshakers_add(exec_ctx, HANDSHAKER_CLIENT, c->args.channel_args,
                        c->handshake_mgr);
+  grpc_endpoint_add_to_pollset_set(exec_ctx, c->endpoint,
```

Comment:
Do we need to also remove this endpoint from interested_parties when handshaking is complete?